code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -191,40 +250,48 @@
 	reagent_state = LIQUID
 	color = "#FAFAFA"
 	chem_flags = CHEMICAL_RNG_GENERAL | CHEMICAL_RNG_FUN | CHEMICAL_RNG_BOTANY | CHEMICAL_GOAL_CHEMIST_DRUG | CHEMICAL_GOAL_CHEMIST_BLOODSTREAM | CHEMICAL_GOAL_BOTANIST_HARVEST
-	overdose_threshold = 20
-	addiction_threshold = 10
-	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	metabolization_rate = REAGENTS_METABOLISM * 0.5
 
 /datum/reagent/drug/methamphetamine/on_mob_metabolize(mob/living/L)
 	..()
 	if(L.client)
 		L.client.give_award(/datum/award/achievement/misc/meth, L)
-
-	L.add_movespeed_modifier(type, update=TRUE, priority=100, multiplicative_slowdown=-0.5, blacklisted_movetypes=(FLYING|FLOATING))
-	ADD_TRAIT(L, TRAIT_SLEEPIMMUNE, type)
 	ADD_TRAIT(L, TRAIT_NOBLOCK, type)
+	addiction_threshold = rand(0, 10) //Highly addictive substances are risky. You never know exactly when you'll be addicted
+	overdose_threshold = rand(10, 20)
```

Comment:
Way too low, especially since it decreases every tick

Reply:
You will never hit overdose or addiction in the middle of metabolization, only immediately after taking more. 

It reduces every tick but by less than the metabolization rate.